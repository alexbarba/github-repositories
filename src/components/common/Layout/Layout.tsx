import { FC, useState } from 'react'
import { Header, Footer } from 'components/common'
import { SunIcon, MoonIcon } from '@primer/octicons-react'

const ThemeSwitch = ({ isDark = true, setIsDark = (val: boolean) => {} }) => (
  <button
    className="fixed z-10 flex items-center justify-center text-white border rounded-full bg-secondary dark:bg-white h-9 w-9 bottom-6 left-6 border-accent-br dark:text-secondary"
    onClick={() => setIsDark(!isDark)}>
    {isDark ? <SunIcon /> : <MoonIcon />}
  </button>
)

const Layout: FC = ({ children }) => {
  const [isDark, setIsDark] = useState(true)

  return (
    <div className={isDark ? 'dark ' : 'light'}>
      <div className="w-screen min-h-screen max-w-[100%] text-primary dark:bg-secondary md:dark:bg-primary ">
        <Header />
        <main
          className="max-w-screen-xl min-h-full px-4 mx-auto md:px-9"
          style={{ scrollbarColor: 'transparent' }}>
          {children}
        </main>
        <Footer />
      </div>
      <ThemeSwitch isDark={isDark} setIsDark={setIsDark} />
    </div>
  )
}

export default Layout
