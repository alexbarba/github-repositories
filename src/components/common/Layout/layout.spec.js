import { render } from '@testing-library/react'
import Layout from './Layout'

describe('Layout', () => {
  it('should be match with snapshot', () => {
    const { asFragment } = render(<Layout />)
    expect(asFragment()).toMatchSnapshot()
  })
})
