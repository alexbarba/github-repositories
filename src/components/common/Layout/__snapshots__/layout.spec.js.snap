// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should be match with snapshot 1`] = `
<DocumentFragment>
  <div
    class="dark "
  >
    <div
      class="w-screen min-h-screen max-w-[100%] text-primary dark:bg-secondary md:dark:bg-primary "
    >
      <header
        class="h-16 md:h-[60px] p-4 lg:px-8 md:px-6 bg-header text-sm text-header"
      >
        <div
          class="flex items-center justify-between md:hidden"
        >
          <svg
            aria-hidden="true"
            class="octicon octicon-three-bars"
            fill="currentColor"
            height="24"
            role="img"
            style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
            viewBox="0 0 16 16"
            width="24"
          >
            <path
              d="M1 2.75A.75.75 0 011.75 2h12.5a.75.75 0 110 1.5H1.75A.75.75 0 011 2.75zm0 5A.75.75 0 011.75 7h12.5a.75.75 0 110 1.5H1.75A.75.75 0 011 7.75zM1.75 12a.75.75 0 100 1.5h12.5a.75.75 0 100-1.5H1.75z"
              fill-rule="evenodd"
            />
          </svg>
          <a
            href="https://github.com"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-mark-github"
              fill="currentColor"
              height="32"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 16 16"
              width="32"
            >
              <path
                d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
          <a
            href="https://github.com/notifications"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-bell"
              fill="currentColor"
              height="16"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 16 16"
              width="16"
            >
              <path
                d="M8 16a2 2 0 001.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 008 16z"
              />
              <path
                d="M8 1.5A3.5 3.5 0 004.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.018.018 0 00-.003.01l.001.006c0 .002.002.004.004.006a.017.017 0 00.006.004l.007.001h10.964l.007-.001a.016.016 0 00.006-.004.016.016 0 00.004-.006l.001-.007a.017.017 0 00-.003-.01l-1.703-2.554a1.75 1.75 0 01-.294-.97V5A3.5 3.5 0 008 1.5zM3 5a5 5 0 0110 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.518 1.518 0 0113.482 13H2.518a1.518 1.518 0 01-1.263-2.36l1.703-2.554A.25.25 0 003 7.947V5z"
                fill-rule="evenodd"
              />
            </svg>
          </a>
        </div>
        <div
          class="items-center justify-between hidden md:flex"
        >
          <div
            class="flex items-center w-full gap-4"
          >
            <svg
              aria-hidden="true"
              class="octicon octicon-mark-github"
              fill="currentColor"
              height="32"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 16 16"
              width="32"
            >
              <path
                d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
                fill-rule="evenodd"
              />
            </svg>
            <label
              class="flex border border-white rounded-md border-opacity-30 h-7 min-w-[15.5rem] max-w-[17rem] px-2"
            >
              <input
                class="box-border w-full h-full py-0 border-none bg-header focus-visible:outline-none"
                placeholder="Search or jump to..."
              />
              <button
                class="btn w-auto h-auto px-2 py-0 border-white border-opacity-30 bg-header rounded-[4px] my-[3px] text-sm"
              >
                / 
              </button>
            </label>
            <ul
              class="flex gap-4 font-bold break-words"
            >
              <li>
                <a
                  href="https://github.com/pulls"
                >
                  Pull Requests
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/issues"
                >
                  Issues
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/marketplace"
                >
                  Marketplace
                </a>
              </li>
              <li>
                <a
                  href="https://github.com/explore"
                >
                  Explore
                </a>
              </li>
            </ul>
          </div>
          <div
            class="flex gap-4"
          >
            <a
              href="https://github.com/notifications"
            >
              <svg
                aria-hidden="true"
                class="octicon octicon-bell"
                fill="currentColor"
                height="16"
                role="img"
                style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M8 16a2 2 0 001.985-1.75c.017-.137-.097-.25-.235-.25h-3.5c-.138 0-.252.113-.235.25A2 2 0 008 16z"
                />
                <path
                  d="M8 1.5A3.5 3.5 0 004.5 5v2.947c0 .346-.102.683-.294.97l-1.703 2.556a.018.018 0 00-.003.01l.001.006c0 .002.002.004.004.006a.017.017 0 00.006.004l.007.001h10.964l.007-.001a.016.016 0 00.006-.004.016.016 0 00.004-.006l.001-.007a.017.017 0 00-.003-.01l-1.703-2.554a1.75 1.75 0 01-.294-.97V5A3.5 3.5 0 008 1.5zM3 5a5 5 0 0110 0v2.947c0 .05.015.098.042.139l1.703 2.555A1.518 1.518 0 0113.482 13H2.518a1.518 1.518 0 01-1.263-2.36l1.703-2.554A.25.25 0 003 7.947V5z"
                  fill-rule="evenodd"
                />
              </svg>
            </a>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="octicon octicon-plus"
                fill="currentColor"
                height="16"
                role="img"
                style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M7.75 2a.75.75 0 01.75.75V7h4.25a.75.75 0 110 1.5H8.5v4.25a.75.75 0 11-1.5 0V8.5H2.75a.75.75 0 010-1.5H7V2.75A.75.75 0 017.75 2z"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="octicon octicon-triangle-down"
                fill="currentColor"
                height="16"
                role="img"
                style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M4.427 7.427l3.396 3.396a.25.25 0 00.354 0l3.396-3.396A.25.25 0 0011.396 7H4.604a.25.25 0 00-.177.427z"
                />
              </svg>
            </div>
            <div
              class="flex items-center"
            >
              <svg
                aria-hidden="true"
                class="border rounded-full"
                fill="currentColor"
                height="20"
                role="img"
                style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                viewBox="0 0 16 16"
                width="20"
              >
                <path
                  d="M10.5 5a2.5 2.5 0 11-5 0 2.5 2.5 0 015 0zm.061 3.073a4 4 0 10-5.123 0 6.004 6.004 0 00-3.431 5.142.75.75 0 001.498.07 4.5 4.5 0 018.99 0 .75.75 0 101.498-.07 6.005 6.005 0 00-3.432-5.142z"
                  fill-rule="evenodd"
                />
              </svg>
              <svg
                aria-hidden="true"
                class="octicon octicon-triangle-down"
                fill="currentColor"
                height="16"
                role="img"
                style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
                viewBox="0 0 16 16"
                width="16"
              >
                <path
                  d="M4.427 7.427l3.396 3.396a.25.25 0 00.354 0l3.396-3.396A.25.25 0 0011.396 7H4.604a.25.25 0 00-.177.427z"
                />
              </svg>
            </div>
          </div>
        </div>
      </header>
      <main
        class="max-w-screen-xl min-h-full px-4 mx-auto md:px-9"
      />
      <footer
        class="flex flex-wrap items-center self-end justify-start px-4 py-10 mt-10 text-xs text-center border-t border-accent-br"
      >
        <div
          class="hidden w-1/4 md:block"
        >
          <span
            class="flex items-center justify-center w-full text-footer"
          >
            <svg
              aria-hidden="true"
              class="mr-2 text-fg-subtle"
              fill="currentColor"
              height="24"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 16 16"
              width="24"
            >
              <path
                d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              class="leading-[1.125rem] align-middle"
            >
              © 2021 GitHub, Inc.
            </span>
          </span>
        </div>
        <ul
          class="flex flex-wrap justify-between w-full gap-x-4 md:w-2/3 text-accent-fg"
        >
          <li>
            <a
              href="https://docs.github.com/en/github/site-policy/github-terms-of-service"
            >
              Terms
            </a>
          </li>
          <li>
            <a
              href="https://docs.github.com/en/github/site-policy/github-privacy-statement"
            >
              Privacy
            </a>
          </li>
          <li>
            <a
              href="https://github.com/security"
            >
              Security
            </a>
          </li>
          <li>
            <a
              href="https://www.githubstatus.com/"
            >
              Status
            </a>
          </li>
          <li>
            <a
              href="https://docs.github.com"
            >
              Docs
            </a>
          </li>
          <li>
            <a
              href="https://support.github.com/?tags=dotcom-footer"
            >
              Contact GitHub
            </a>
          </li>
          <li>
            <a
              href="https://github.com/pricing"
            >
              Pricing
            </a>
          </li>
          <li>
            <a
              href="https://docs.github.com"
            >
              API
            </a>
          </li>
          <li>
            <a
              href="https://services.github.com/"
            >
              Training
            </a>
          </li>
          <li>
            <a
              href="https://github.blog/"
            >
              Blog
            </a>
          </li>
          <li>
            <a
              href="https://github.com/about"
            >
              About
            </a>
          </li>
        </ul>
        <div
          class="w-full mt-2 mb-10 md:hidden"
        >
          <span
            class="flex items-center justify-center w-full text-footer"
          >
            <svg
              aria-hidden="true"
              class="mr-2 text-fg-subtle"
              fill="currentColor"
              height="24"
              role="img"
              style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
              viewBox="0 0 16 16"
              width="24"
            >
              <path
                d="M8 0C3.58 0 0 3.58 0 8c0 3.54 2.29 6.53 5.47 7.59.4.07.55-.17.55-.38 0-.19-.01-.82-.01-1.49-2.01.37-2.53-.49-2.69-.94-.09-.23-.48-.94-.82-1.13-.28-.15-.68-.52-.01-.53.63-.01 1.08.58 1.23.82.72 1.21 1.87.87 2.33.66.07-.52.28-.87.51-1.07-1.78-.2-3.64-.89-3.64-3.95 0-.87.31-1.59.82-2.15-.08-.2-.36-1.02.08-2.12 0 0 .67-.21 2.2.82.64-.18 1.32-.27 2-.27.68 0 1.36.09 2 .27 1.53-1.04 2.2-.82 2.2-.82.44 1.1.16 1.92.08 2.12.51.56.82 1.27.82 2.15 0 3.07-1.87 3.75-3.65 3.95.29.25.54.73.54 1.48 0 1.07-.01 1.93-.01 2.2 0 .21.15.46.55.38A8.013 8.013 0 0016 8c0-4.42-3.58-8-8-8z"
                fill-rule="evenodd"
              />
            </svg>
            <span
              class="leading-[1.125rem] align-middle"
            >
              © 2021 GitHub, Inc.
            </span>
          </span>
        </div>
      </footer>
    </div>
    <button
      class="fixed z-10 flex items-center justify-center text-white border rounded-full bg-secondary dark:bg-white h-9 w-9 bottom-6 left-6 border-accent-br dark:text-secondary"
    >
      <svg
        aria-hidden="true"
        class="octicon octicon-sun"
        fill="currentColor"
        height="16"
        role="img"
        style="display: inline-block; user-select: none; vertical-align: text-bottom; overflow: visible;"
        viewBox="0 0 16 16"
        width="16"
      >
        <path
          d="M8 10.5a2.5 2.5 0 100-5 2.5 2.5 0 000 5zM8 12a4 4 0 100-8 4 4 0 000 8zM8 0a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0V.75A.75.75 0 018 0zm0 13a.75.75 0 01.75.75v1.5a.75.75 0 01-1.5 0v-1.5A.75.75 0 018 13zM2.343 2.343a.75.75 0 011.061 0l1.06 1.061a.75.75 0 01-1.06 1.06l-1.06-1.06a.75.75 0 010-1.06zm9.193 9.193a.75.75 0 011.06 0l1.061 1.06a.75.75 0 01-1.06 1.061l-1.061-1.06a.75.75 0 010-1.061zM16 8a.75.75 0 01-.75.75h-1.5a.75.75 0 010-1.5h1.5A.75.75 0 0116 8zM3 8a.75.75 0 01-.75.75H.75a.75.75 0 010-1.5h1.5A.75.75 0 013 8zm10.657-5.657a.75.75 0 010 1.061l-1.061 1.06a.75.75 0 11-1.06-1.06l1.06-1.06a.75.75 0 011.06 0zm-9.193 9.193a.75.75 0 010 1.06l-1.06 1.061a.75.75 0 11-1.061-1.06l1.06-1.061a.75.75 0 011.061 0z"
          fill-rule="evenodd"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
